packages/server/src/service/alloy.ts

```diff
@@ -0,0 +1,85 @@
+import { processEnvOrThrow, StateInfo } from '../common'
+import axios, { AxiosResponse } from 'axios'
+
+// Env & axios refactoring
+
+const alloyRelaxed = processEnvOrThrow('ALLOY_RELAXED')
+const apiKey = processEnvOrThrow('ALLOY_API_KEY')
+const apiSecret = processEnvOrThrow('ALLOY_API_SECRET')
+const apiUrl = 'https://api.alloy.us/v1'
+const commonAxiosSettings = {
+  auth: {username: apiKey, password: apiSecret},
+  withCredentials: true,
+}
+
+/** Converts StateInfo to URL query params */
+const stateInfoToParams = (info: StateInfo) => {
+  const nameSplit = info.name.split(' ')
+  const firstName = nameSplit[0]
+  const lastName = nameSplit[nameSplit.length - 1]
+  const { city, stateAbbr: state, street, streetNumber, postcode: zip } = info.address
+  const address = `${streetNumber} ${street}`
+
+  return encodeURI(
+    [
+      `first_name=${firstName}`,
+      `last_name=${lastName}`,
+      `address=${address}`,
+      `city=${city}`,
+      `state=${state}`,
+      `zip=${zip}`,
+    ].join('&')
+  )
+}
+
+// Contains only fields used by VoteByMail
+interface AlloyResponse {
+  data: {
+    // There are other possible registration_status, but for our interests
+    // only to know if !== 'Active' is enough
+    registration_status: 'Active' | null
+  }
+}
+
+class Alloy {
+  /**
+   * Returns true if a voter can vote, false otherwise.
+   */
+  public isRegistered = async (voter: StateInfo): Promise<boolean> => {
+    const url = `${apiUrl}/verify?${stateInfoToParams(voter)}`
+    let response: AxiosResponse<AlloyResponse>
+    let attempts = 0
```

Comment:
Consider using https://www.npmjs.com/package/axios-retry or https://www.npmjs.com/package/retry-axios instead of building your own.  Also, you'll want to exponentially backoff between submissions, which you can do using these libraries.